tasklist/client/src/modules/queries/useCurrentUser.tsx

```diff
@@ -15,20 +15,21 @@ function useCurrentUser() {
   return useQuery<CurrentUser, RequestError | Error>({
     queryKey: ['currentUser'],
     queryFn: async () => {
-      const {response, error} = await request(api.v1.getCurrentUser());
+      const {response, error} = await request(api.v2.getCurrentUser());
 
       if (response !== null) {
         const currentUser = await response.json();
-        const permissions: unknown[] = currentUser?.permissions ?? [];
-
+        const permissions: unknown[] | undefined = currentUser?.permissions;
         return {
           ...currentUser,
-          permissions: permissions
-            .filter<string>(
-              (permission): permission is string =>
-                typeof permission === 'string',
-            )
-            .map((permission) => permission.toLowerCase()),
+          permissions:
+            permissions &&
```

Comment:
@vsgoulart I checked on reverting this too, but intellij indicates this as issue as permission could be undefined, would keep it though?
![Uploading image.png…]()

Reply:
but actually it shouldn't be possible to be undefined given we add ` ?? [];`, right? Apologies, my TS knowledge is next to zero :D

Reply:
Yeah, that's the issue

The explicit typing for undefined is the issue

Reply:
got it, will revert that properly